### PR TITLE
Adding a new feature : Notification

### PR DESCRIPTION
**Pull request by AntonRagot**
This PR includes:

- The addition of the notification feature (slight change in the DB to include the notification key, a class to create the JSON that is going to be send and a class to create the notification from the message received)

- The corresponding mock

What is missing: 

- A few check in the Firebase notification to control the ReceivedMessage

- A test for the class Firebase notification (but is tricky to implement - hence the PR)

Additional stuff to do (if I have the time):

- Create a toggle button in settings to allow/disallow notifications

### CODE REVIEW

**Comment by somecookie** [comment]
A suggestion for another feature that could be implemented by one of us if we have the time: notification when someone follows us

File: app/src/main/java/ch/epfl/sweng/runpharaa/notification/FireMessage.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class FireMessage {
+    //Server key found in firebase
+    private final String SERVER_KEY = "AAAArozt-xs:APA91bEN2m2cVZVUlJ6GMRgacY42W7KF9f3Orpm1bcEKF0WbugfgDfwcKLZ1qmgs4d4EmD7iCgkB3qH6oXfjEk0vQEg2oodWioeYzve__QeCJr2PPgLI4qmHi71L5_EuxL87y1TJNlpX";
```
Comment (somecookie) [inline app/src/main/java/ch/epfl/sweng/runpharaa/notification/FireMessage.java]: We should think about something to avoid having the API keys in the code

**Comment by AntonRagot** [comment]
> A suggestion for another feature that could be implemented by one of us if we have the time: notification when someone follows us and when a comment is added to our track or the track we have commented

Good idea!
Doing it is two lines so why not, but the mock might take a bit more... I'll try

File: app/src/main/java/ch/epfl/sweng/runpharaa/notification/FireMessage.java (hunk 1 above)
Comment (PeterKrcmar0) [reply app/src/main/java/ch/epfl/sweng/runpharaa/notification/FireMessage.java]: Agreed

File: app/src/main/java/ch/epfl/sweng/runpharaa/utils/Util.java [hunk 2]
```diff
@@ -110,32 +110,7 @@ static int calculateInSampleSize(
         return inSampleSize;
     }
 
-    /*static Intent getTwitterIntent(Context ctx, String shareText) {
```
Comment (PeterKrcmar0) [inline app/src/main/java/ch/epfl/sweng/runpharaa/utils/Util.java]: Thanks for removing this, the less useless commented code we have, the better.

File: app/src/main/java/ch/epfl/sweng/runpharaa/TrackPropertiesActivity.java [hunk 3]
```diff
@@ -287,12 +290,21 @@ else if (i == nbrTypes - 2)
...
         final Track track = track1;
         if (User.instance.alreadyLiked(trackID)) {
             track.getProperties().removeLike();
             User.instance.unlike(trackID);
             UserDatabaseManagement.removeLikedTrack(trackID);
         } else {
+            UserDatabaseManagement.getNotificationKeyFromUID(track.getCreatorUid(), new Callback<String>() {
+                @Override
+                public void onSuccess(String value) {
+                    sentToNotification(value, "LIKE ALERT", "The user " + User.instance.getName() + " liked your track " + track.getName());
```
[the remark this answers is not in the document]
Comment (QlimaxXael) [reply app/src/main/java/ch/epfl/sweng/runpharaa/TrackPropertiesActivity.java]: Small tip: you may already know this but I learned recently that adding hard-coded strings to the "strings" file is as easy as pressing alt-enter and choosing "add to strings file" 👍

File: app/src/androidTest/java/ch/epfl/sweng/runpharaa/UsersProfileActivityTest.java [hunk 4]
```diff
@@ -184,4 +187,17 @@ private void clickOnDifferenttrophies(){
         sleep(WAIT_TIME);
         Espresso.pressBack();
     }
+
+    @Test
+    public void testWriteUser(){
```
[the remark this answers is not in the document]
Comment (AntonRagot) [reply app/src/androidTest/java/ch/epfl/sweng/runpharaa/UsersProfileActivityTest.java]: I changed it to login activity test

**Pull request merged**